cmd/notification-service/main.go

```diff
@@ -399,18 +408,42 @@ func wmBrokerConfiguration(conf config.Configuration, logger *slog.Logger, metri
 	}
 }
 
-func initEventPublisherDriver(ctx context.Context, logger *slog.Logger, conf config.Configuration, metricMeter metric.Meter) (message.Publisher, error) {
-	provisionTopics := []watermillkafka.AutoProvisionTopic{}
-	if conf.Notification.Consumer.DLQ.AutoProvision.Enabled {
-		provisionTopics = append(provisionTopics, watermillkafka.AutoProvisionTopic{
-			Topic:         conf.Notification.Consumer.DLQ.Topic,
-			NumPartitions: int32(conf.Notification.Consumer.DLQ.AutoProvision.Partitions),
-			Retention:     conf.BalanceWorker.DLQ.AutoProvision.Retention,
+func initEventPublisherDriver(ctx context.Context, logger *slog.Logger, conf config.Configuration, metricMeter metric.Meter, topicProvisioner pkgkafka.TopicProvisioner) (message.Publisher, error) {
+	var provisionTopics []pkgkafka.TopicConfig
+	if conf.BalanceWorker.DLQ.AutoProvision.Enabled {
```

Comment:
This is the notification service, isn't it?

Reply:
Nice catch! Will fix it.